### PR TITLE
Sustainable validate

### PR DESCRIPTION
**Pull request by EliasEriksson**
* renamed cap to limit for consistency
* providing limit in allocated will give a multiple of `cadence.value` that is less than or equal to the limit
* call signature of functions with optional date is changed to make room for multiple optional parameters
* validate function utilizes sustainable days instead of calculating balances.

### CODE REVIEW

File: Cadence/index.ts [hunk 1]
```diff
@@ -44,7 +44,9 @@ export namespace Cadence {
 			else
 				result = cadence.value
 		}
-		return options?.cap == undefined ? result : Math.min(options.cap, result)
+		return options?.limit == undefined
+			? result
+			: Math.trunc(Math.min(options.limit, result) / cadence.value) * cadence.value
```
Comment (linusarnejohansson) [inline Cadence/index.ts:49]: why do we divide by cadence.value, truncate the result and then multiply with cadence.value again?

File: Cadence/index.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export namespace Cadence {
 	})
 	export const is = type.is
 	export const flaw = type.flaw
-	export function allocated(cadence: Cadence, date: isoly.Date, options?: { cap?: number }): number {
+	export function allocated(cadence: Cadence, date: isoly.Date, options?: { limit?: number }): number {
```
Comment (linusarnejohansson) [inline Cadence/index.ts:24]: I think we are going to need some comments explaining what the different parameters actually mean. For example, what date is it that we are giving the function?

File: Cadence/index.ts [hunk 3]
```diff
@@ -55,10 +57,10 @@ export namespace Cadence {
 	/**
 	 * formula described here https://github.com/issuefab/app/issues/232
```
Comment (linusarnejohansson) [inline Cadence/index.ts:58]: Do we want to add the formula in the comments or are we fine with keeping it on github?

How is that going to work when/if we switch to github.com/proquse?

File: Cadence/index.ts (hunk 1 above)
Comment (linusarnejohansson) [reply Cadence/index.ts:49]: comment for line 41 and 43:  Do we handle year stuff  when the cadence is 'week' and 'day'?

File: Cadence/index.ts (hunk 2 above)
Comment (linusarnejohansson) [reply Cadence/index.ts:24]: consider the same for all of the other functions that we have in this file that handles how sustainable something is

**Review by linusarnejohansson — CHANGES_REQUESTED** [review]

File: Cadence/index.ts (hunk 2 above)
Comment (EliasEriksson) [reply Cadence/index.ts:24]: date can not really mean many things here. it should be strait forward.

File: Cadence/index.ts (hunk 1 above)
Comment (EliasEriksson) [reply Cadence/index.ts:49]: magic

Comment (linusarnejohansson) [reply Cadence/index.ts:49]: seems reasonable

**Review by linusarnejohansson — APPROVED** [review]

**Pull request merged**